newIDE/app/public/external/jfxr/jfxr-main.js

```diff
@@ -51,17 +51,26 @@ const saveSoundEffect = pathEditor => {
   });
 };
 
-// Wait for the window to be fully initialized before sending the
-// ready event. Don't use DOMContentLoaded as it was observed to be fired
-// even if jfxr DOM/scripts are not yet loaded.
+// Repeatedly try to gain access to jfxr's control element and its methods
+// When succeeding, stop trying. Aggresive approach, makes sure analytics slowing 
+// the load process of the iframe doesnt affect the editor.
 const editorFrameEl = document.getElementById('jfxr-frame');
-editorFrameEl.onload = () =>{
-  ipcRenderer.send('jfxr-ready');
+const tryToGetJsfx = () => {
+  if (jfxr === null) {
+    editorContentDocument = editorFrameEl.contentDocument;
+    jfxr = editorFrameEl.contentWindow.angular
+      .element(editorContentDocument.getElementsByClassName('ng-scope')[0])
+      .scope().ctrl;
+  } else { // gained access to control elements!
```

Comment:
This else looks strange.
If I read this function, it says "when you tryToGetJsfx, you check if jfxr is null, if yes you find get the reference ELSE you say it's ready".

I think it should be:

```js
const tryToGetJsfx = () => {
  editorContentDocument = editorFrameEl.contentDocument;
  jfxr = editorFrameEl.contentWindow.angular
      .element(editorContentDocument.getElementsByClassName('ng-scope')[0])
      .scope().ctrl;
  if (jfxr !== null) {
    ipcRenderer.send('jfxr-ready');
    clearInterval(retryToGetJsfx);
  }
};
```